src/odrive_can_node.cpp

```diff
@@ -191,6 +216,17 @@ void ODriveCanNode::request_state_callback() {
     can_intf_.send_can_frame(frame);
 }
 
+void ODriveCanNode::request_clear_errors_callback() {
+    struct can_frame frame;
+    frame.can_id = node_id_ << 5 | CmdId::kClearErrors;
+    {
+        std::unique_lock<std::mutex> guard(axis_state_mutex_);
+        write_le<uint8_t>(identify_, frame.data);
```

Comment:
Ah I don't mean autoclear during the run (while the motor is active). What I mean is, your ROS application explicitly requests a state change from IDLE to CLOSED_LOOP_CONTROL to enable the motor. Since this request is explicit, it means that by this time your application had an opportunity to read potential errors and to decide that it's ok to re-enable the motor. Therefore I think it makes sense for the ROS node to clear errors by itself on this transition.
Once the ODrive is in CLOSED_LOOP_CONTROL, it will go back to IDLE by itself as soon as there's a new error.

In addition, on the ODrive S1, clearing errors makes sure the brake resistor is active (in case it previously got disarmed due to undervoltage). Requesting CLOSED_LOOP_CONTROL otherwise causes a `BRAKE_RESISTOR_DISARMED` error. To avoid this error, I think in this case it's especially nice if the ROS node sends ClearErrors by itself.

Reply:
Neither did I. Actually the break resistor is good example, when the state change command implicitly clears errors, you would newer know about the undervoltage, with explicit call to clear errors you can chose to ignore it or somehow deal with it...

Reply:
In both cases the user can check errors before entering CLOSED_LOOP_CONTROL, and handle them accordingly. The usage would be like this:

- Without auto-clearing in ROS node:
  1. Check errors
  2. Clear errors
  3. Request CLOSED_LOOP_CONTROL
- With auto-clearing in ROS node:
  1. Check errors
  2. Request CLOSED_LOOP_CONTROL

I think what you might be referring to is more about educating the user, rather than client code capability: An uninformed user might indeed miss errors in the second variant, but an informed user can read them in both cases.

However regarding that, I should clarify the semantics of errors and ClearErrors: The error variable `disarm_reason` does _not_ prevent the ODrive itself from re-entering CLOSED_LOOP_CONTROL. The only reasons to call ClearErrors are:
- re-enable brake resistor (S1 only)
- be sure that when `disarm_reason` indicates an error at a later time, it was newer than the last call to ClearErrors
- make the LED not be red

This change in semantics was a conscious choice that dates back to the early days of `odrivetool`: Everyone just got used to calling `odrv0.clear_errors()` before `odrv0.axis0.requested_state = AxisState.CLOSED_LOOP_CONTROL`, without even looking at the errors. So we decided to do it automatically on the ODrive, but just missed the three points above.